x-pack/plugins/data_enhanced/server/search/session/monitoring_task.ts

```diff
@@ -29,7 +32,8 @@ interface SearchSessionTaskDeps {
 function searchSessionRunner(
   core: CoreSetup<DataEnhancedStartDependencies>,
   { logger, config }: SearchSessionTaskDeps
-) {
+): TaskRunCreatorFunction {
+  const aborted$ = new Subject<void>();
```

Comment:
Since we have one global `aborted$` for all instances of this task, doesn't this mean that if one instance hits the timeout, any others in-progress will also be aborted? What if we moved the initialization of `aborted$` between lines 37/38?

Reply:
I think this looks a bit confusing because of how TASK definition API is designed,
you have to implement `run` and `cancel` function 

```
run() { },
cancel() { }
```

This means you need somehow outside of the`run` to have a reference of what you want to cancel. 

With the current API _I think_ `abort$` approach works correctly:

I **assume** that API is called sequentially and `run`s can't overlap: 

```
- run()
- cancel()
- run()
- cancel() 
```

In this case, calling `abort$` inside `cancel` would abort previous running tasks.

I would like @elastic/kibana-alerting-services to review how I use their API 🙏

Reply:
This doesn't look global to me, actually.
A TaskRunner is creates whenever a task is picked up by Kibana, so each task will have its own abort observable, which will be GCed when that task completes.

Looking at this usage, this PR LGTM 👍